components/up-job.libsonnet

```diff
@@ -73,6 +73,79 @@ local k = import 'ksonnet/ksonnet.beta.4/k.libsonnet';
     },
   },
 
+  withGetToken:: {
+    local job = k.batch.v1.job,
+    local container = job.mixin.spec.template.spec.containersType,
+    local u = self,
+    config+:: {
+      curlImage: error 'must provide image for cURL',
+      tokenEndpoint: error 'must provide token endpoint',
+      username: error 'must provide username',
+      password: error 'must provide password',
+      clientID: error 'must provide clientID',
+      clientSecret: error 'must provide clientSecret',
+    },
+
+    job+: {
+      spec+: {
+        template+: {
+          spec+: {
+            local c =
+              container.new('curl', u.config.curlImage) +
+              container.withCommand([
+                '/bin/sh',
+                '-c',
+                |||
+                  curl --request POST \
```

Comment:
Isn't this something that can be added directly to the up tool to avoid this extra dependency and simplify the config?

Reply:
I don't think we should. this is getting an OIDC token via the password_grant flow, meaning we would have to make the up tool an OAuth client application, which is more complexity than I think that tool should have

Reply:
Agreed with squat